chroma_core/models/client_mount.py

```diff
@@ -293,7 +293,17 @@ def description(self):
     def get_steps(self):
         search = lambda cm: (cm.host == self.host and cm.state == "unmounted")
         unmounted = ObjectCache.get(LustreClientMount, search)
-        args = dict(host=self.host, filesystems=[(m.filesystem.mount_path(), m.mountpoint) for m in unmounted])
+        args = {
+            "host": self.host,
+            "filesystems": [
+                (
+                    ObjectCache.get_one(ManagedFilesystem, lambda mf, mtd=m: mf.name == mtd.filesystem).mount_path(),
```

Comment:
What's the difference comparing to the above two lines (`search=` and `unmounted=`)?

Reply:
The `unmounted` is a list of unmount `LustreClientMount` objects. We loop through each of these and match on the filesystem whose name in the umounted client mount objects matches the filesystem name. That will then give a list of filesystems that correspond to the mounts. For example, say we have:

Filesystems:
fs1, fs2, and fs3.

Mounts:
mnt1, mnt2

mnt1 is associated with fs1
mnt2 is associated with fs2

We unmount both of these instances but their object still exists. We then later want to mount them. The expression would return a list with:
[fs1, fs2]

Reply:
The problem i'm working through at the moment is that the mount path is erased when a client is unmounted.

Reply:
> That can (should) be in a separate patch.